src/biocframe/BiocFrame.py

```diff
@@ -1019,6 +1032,68 @@ def remove_columns(self, columns: Sequence[Union[int, str]], in_place: bool = Fa
 
         return output
 
+    def remove_row(self, row: Union[int, str], in_place: bool = False) -> "BiocFrame":
+        """Remove a row. This is a convenience wrapper around :py:attr:`~remove_rows`.
+
+        Args:
+            row:
+                Name or positional index of the row to remove.
+
+            in_place:
+                Whether to modify the object in place. Defaults to False.
+
+        Returns:
+            A modified ``BiocFrame`` object, either as a copy of the original
+            or as a reference to the (in-place-modified) original.
+        """
+        return self.remove_rows([row], in_place=in_place)
+
+    def remove_rows(self, rows: Sequence[Union[int, str]], in_place: bool = False) -> "BiocFrame":
+        """Remove any number of existing rows.
+
+        Args:
+            rows:
+                Names or indices of the rows to remove.
+
+            in_place:
+                Whether to modify the object in place. Defaults to False.
+
+        Returns:
+            BiocFrame: A modified ``BiocFrame`` object, either as a copy of the original
+            or as a reference to the (in-place-modified) original.
+        """
+        output = self._define_output(in_place)
+        if not in_place:
+            output._data = copy(output._data)
+
+        killset = set()
+        for name in rows:
+            if isinstance(name, int):
+                name = output._row_names[name]
+            if name not in output._row_names:
+                raise ValueError(f"Row '{name}' does not exist.")
+            killset.add(name)
```

Comment:
Correct me if i am wrong, Reading this, I think it supports cases where `rows` can be a combination of both names and indices. We should avoid this. This should expect either all values in rows to be either strings (for row_names) or indices for positions. Ideally also supports slice objects to keep.

Reply:
Can you also add tests for this?